apps/jubmoji-quest/src/pages/cardholder/index.tsx

```diff
@@ -0,0 +1,486 @@
+                }
+              >
+                <PowerCard
+                  title="Pseudonymous X post"
+                  description="Cardholders can post to X using their jubmoji identity"
+                  powerType="TWITTER"
+                />
+              </div> */}
+
+              <div
+              // onClick={() =>
+              //   setCardholderAction(CardholderActions.PSEUD_TELEGRAM_POST)
+              // }
+              >
+                <PowerCard
+                  title="Pseudonymous TG (Soon™)"
+                  description="Cardholders can message on TG using their jubmoji identity"
+                  powerType="TELEGRAM"
+                  disabled
+                />
+              </div>
+
+              <Message>{`Cardholder gated (need 1 tap)`}</Message>
+
+              <div
+                onClick={() =>
+                  setCardholderAction(CardholderActions.PROPOSE_QUEST)
+                }
+              >
+                <PowerCard
+                  title="Propose a quest"
+                  description="Cardholders can propose quests for the community"
+                  powerType="REDIRECT"
+                />
+              </div>
+
+              <div
+                onClick={() =>
+                  setCardholderAction(CardholderActions.JOIN_TG_GROUP)
+                }
+              >
+                <PowerCard
+                  title="Join cardholder TG group"
+                  description="Group where updates are sent & feedback can be shared"
+                  powerType="TELEGRAM"
+                />
+              </div>
+
+              <Message>{`Open to all`}</Message>
+
+              <Link href="/card-request">
+                <PowerCard
+                  title="Apply to be a cardholder"
+                  description="Describe what you'd like to do with your card during Devconnect"
+                  powerType="QR_CODE"
+                />
+              </Link>
+            </div>
+          </div>
+        </div>
+      </div>
+    </>
+  );
+}
+
+enum CardholderTapModalState {
+  UNLOCKING,
+  EDITING,
+  CONFIRMING,
+}
+
+interface CardholderTapModalProps extends ModalProps {
+  action: CardholderActions;
+}
+
+const CardholderTapModal = ({
+  action,
+  isOpen,
+  setIsOpen,
+}: CardholderTapModalProps) => {
+  const md = useRef<any>();
+  useEffect(() => {
+    md.current = new MobileDetect(window?.navigator?.userAgent);
+  }, []);
+  const deviceImage = md?.current?.is("iPhone")
+    ? "/images/tap-phone-ios.png"
+    : "/images/tap-phone-android.png";
+
+  const [modalState, setModalState] = useState(
+    CardholderTapModalState.UNLOCKING
+  );
+  const [pubKeyIndex, setPubKeyIndex] = useState<number>();
+  const [meaning, setMeaning] = useState<string>();
+  const [telegramLink, setTelegramLink] = useState<string>();
+  const [website, setWebsite] = useState<string>("");
+
+  const initialTap = async () => {
```

Comment:
Lots of shared code with `ForegroundTapModal` here. We can duplicate this for now to unblock, but should track this somewhere.